File: tv/src/main/java/com/synergics/stb/iptv/leanback/models/db/TVItemDao.java
package com.synergics.stb.iptv.leanback.models.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.synergics.stb.iptv.leanback.models.TVItems;

import java.util.List;

@Dao
public interface TVItemDao {
    @Query("SELECT * FROM tv_items")
    List<TVItems> getAll();

    @Query("SELECT * FROM tv_items WHERE categoryId=:categoryId")
    List<TVItems> getByCategory(int categoryId);

    @Query("SELECT * FROM tv_items WHERE id=:id")
    TVItems getById(int id);

    @Insert
    void insert(TVItems... items);

    @Insert
    void insert(List<TVItems> items);

    @Update
    void update(TVItems... items);

    @Delete
    void delete(TVItems... items);
}

File: tv/src/main/java/com/synergics/stb/iptv/leanback/models/db/TVItemDao.kt
package com.synergics.stb.iptv.leanback.models.db

import androidx.room.*
import com.synergics.stb.iptv.leanback.models.TVItems

@Dao
interface TVItemDao {
    @get:Query("SELECT * FROM tv_items")
    val all: List<TVItems?>?

    @Query("SELECT * FROM tv_items WHERE categoryId=:categoryId")
    fun getByCategory(categoryId: Int): List<TVItems>

    @Query("SELECT * FROM tv_items WHERE id=:id")
    fun getById(id: Int): TVItems?

    @Insert
    fun insert(vararg items: TVItems?)

    @Insert
    fun insert(items: List<TVItems>?)

    @Update
    fun update(vararg items: TVItems?)

    @Delete
    fun delete(vararg items: TVItems?)

    @Query("delete from tv_items")
    fun clear()

    @Query("delete from sqlite_sequence where name='tv_items'")
    fun resetAI()
}